appserver/ejb/ejb-container/src/main/java/org/glassfish/ejb/deployment/node/runtime/EjbNode.java

```diff
@@ -136,80 +138,81 @@ public void setElementValue(XMLElement element, String value) {
             }
             if (descriptor==null) {
                 DOLUtils.getDefaultLogger().log(Level.SEVERE, "enterprise.deployment.backend.addDescriptorFailure",
-                            new Object[]{element , value}); 
+                        new Object[]{element , value}); 
             } else {
                 if (availEnabled != null) {
                     descriptor.getIASEjbExtraDescriptors().setAttributeValue(IASEjbExtraDescriptors.AVAILABILITY_ENABLED, availEnabled);
                 }
             }
             return;
-        } else if (descriptor==null && ! RuntimeTagNames.AVAILABILITY_ENABLED.equals(element.getQName())) {
+        } else if (descriptor==null && !isAvailabilityEnabled) {
             DOLUtils.getDefaultLogger().log(Level.SEVERE, "enterprise.deployment.backend.addDescriptorFailure",
-                            new Object[] {element.getQName() , value });
+                    new Object[] {qName , value });
+            super.setElementValue(element, value);
+            return;
```

Comment:
The issue was that `descriptor` can be null what is not handled by below if-else-if block. If `qName` is set to do `isAvailabilityEnabled` no descriptor is involved so no warning was logged and no NPE can occur. But if it is some other `qName` we assume a `descriptor` is needed and log the warning. To void the NPE in case one of the `else if` matches we do return. As the default for no match (last `else`) is to call `super` we do this before returning assuming it could be any other `qName` that does not need a `descriptor`.